src/makeInfinityAdder.js

```diff
@@ -6,7 +6,25 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let totalSum = 0;
+
+  const adder = number => {
+    if (number === undefined) {
+      const result = totalSum;
+
+      totalSum = 0;
+
+      return result;
+    }
+
+    totalSum += number;
+
+    return function(nextNumber) {
+      return adder(nextNumber);
+    };
```

Comment:
Why do you use this wrapper? All you need is to return the reference to your 'adder' function

Reply:
Yes, turns out it is much more simple solution. Thanks.